public/js/deleteCourseModal.js

```diff
@@ -0,0 +1,59 @@
+let deleteButtons = document.querySelectorAll('.btnDeleteCourse');
+for (let db of deleteButtons) {
+    db.addEventListener("click", () => {
+        const courseId = db.getAttribute('data-courseId');
+        displayDetails('./api/deleteCourse/' + courseId, courseId);
+    });
+}
+
+function displayDetails(url, courseId) {
+    fetch(url).then(
+        response => {
+            return response.json();
+        }).then(data => {
+            let studentNamesContainer = document.querySelector('#studentNamesContainer');
+            studentNamesContainer.innerHTML = '<h4>Name</h4>';
+            let DeleteCoursesForm = document.querySelector('#DeleteCoursesForm');
+            DeleteCoursesForm.innerHTML = '';
+            DeleteCoursesForm.action = "api/deleteCourse/" + courseId;
```

Comment:
Isn't this pretty much the same as the `url`? So maybe `courseId` isn't needed at all?